### PR TITLE
Add spam protection to fountain

### PR DESCRIPTION
**Pull request by Kubuxu**
Uses reCAPTCHAv3, `RECAPTCHA_SITE_KEY` and `RECAPTCHA_SECRET_KEY` need to be
set in env.

### CODE REVIEW

**Comment by Kubuxu** [comment]
@travisperson I suggest starting with a low threshold (0.1 maybe). Then we can observe how it looks like.

**Comment by Kubuxu** [comment]
CR PLS

**Review by travisperson — APPROVED** [review]

File: cmd/lotus-fountain/main.go [hunk 1]
```diff
@@ -123,22 +130,63 @@ var runCmd = &cli.Command{
...
 type handler struct {
 	ctx context.Context
 	api api.FullNode
 
 	from           address.Address
 	sendPerRequest types.FIL
 
-	limiter *Limiter
+	limiter       *Limiter
+	recapTreshold float64
```
Comment (arajasek) [inline cmd/lotus-fountain/main.go:152]: typo

Comment (arajasek) [inline cmd/lotus-fountain/main.go:152]:
```suggestion
	recapThreshold float64
```

File: cmd/lotus-fountain/main.go [hunk 2]
```diff
@@ -107,11 +112,13 @@ var runCmd = &cli.Command{
 				WalletRate:  15 * time.Minute,
 				WalletBurst: 2,
 			}),
+			recapTreshold: cctx.Float64("captcha-threshold"),
```
Comment (arajasek) [inline cmd/lotus-fountain/main.go:115]:
```suggestion
			recapThreshold: cctx.Float64("captcha-threshold"),
```

File: cmd/lotus-fountain/main.go [hunk 3]
```diff
@@ -123,22 +130,63 @@ var runCmd = &cli.Command{
...
+		}
+		reqIP = h
+	}
+
+	capResp, err := VerifyToken(r.FormValue("g-recaptcha-response"), reqIP)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadGateway)
+		return
+	}
+	if !capResp.Success || capResp.Score < h.recapTreshold {
```
Comment (arajasek) [inline cmd/lotus-fountain/main.go:175]:
```suggestion
	if !capResp.Success || capResp.Score < h.recapThreshold {
```

**Review by arajasek — APPROVED** [review]

**Review by magik6k — APPROVED** [review]

**Pull request merged**